sphinxcontrib/doxylink/parsing.py

```diff
@@ -94,11 +94,13 @@ def normalise(symbol: str) -> Tuple[str, str]:
 
     # This is a very common signature so we'll make a special case for it. It requires no parsing anyway
     if arglist_input_string.startswith('()'):
-        arglist_input_string_no_spaces = arglist_input_string.replace(' override', '').replace(' final', '').replace(' ', '')
-        if arglist_input_string_no_spaces in ('()', '()=0', '()=default'):
-            return function_name, '()'
-        if arglist_input_string_no_spaces in ('()const', '()const=0'):
-            return function_name, '() const'
+        arglist_input_string_no_spaces = (arglist_input_string.replace(' override', '').replace(' final', '')
+                                          .replace(' ', '').replace('=0', '').replace('=default', ''))
+        exception_qualifier = max(arglist_input_string_no_spaces.find('noexcept'),
+                                  arglist_input_string_no_spaces.find('throw'))
+        arglist_input_string_no_spaces = arglist_input_string_no_spaces[
+                                         :exception_qualifier] if exception_qualifier >= 0 else arglist_input_string_no_spaces
```

Comment:
```suggestion
        substrings_to_remove = (' override', ' final', ' ', '=0', '=default')
        for part in substrings_to_remove:
            arglist_input_string_no_spaces = arglist_input_string_no_spaces.replace(part, '')
        exception_qualifier = max(arglist_input_string_no_spaces.find('noexcept'),
                                  arglist_input_string_no_spaces.find('throw'))
        if exception_qualifier != -1:
            # Remove everything starting with the exception keyword
            arglist_input_string_no_spaces = arglist_input_string_no_spaces[:exception_qualifier]
```
Refactored for improved readability and maintainability.